tests/test_model_serializer.py

```diff
@@ -169,53 +170,52 @@ class Meta:
                 model = RegularFieldsModel
                 fields = '__all__'
 
-        expected = dedent("""
-            TestSerializer():
-                auto_field = IntegerField(read_only=True)
-                big_integer_field = IntegerField()
-                boolean_field = BooleanField(default=False, required=False)
-                char_field = CharField(max_length=100)
-                comma_separated_integer_field = CharField(max_length=100, validators=[<django.core.validators.RegexValidator object>])
-                date_field = DateField()
-                datetime_field = DateTimeField()
-                decimal_field = DecimalField(decimal_places=1, max_digits=3)
-                email_field = EmailField(max_length=100)
-                float_field = FloatField()
-                integer_field = IntegerField()
-                null_boolean_field = BooleanField(allow_null=True, default=False, required=False)
-                positive_integer_field = IntegerField()
-                positive_small_integer_field = IntegerField()
-                slug_field = SlugField(allow_unicode=False, max_length=100)
-                small_integer_field = IntegerField()
-                text_field = CharField(max_length=100, style={'base_template': 'textarea.html'})
-                file_field = FileField(max_length=100)
-                time_field = TimeField()
-                url_field = URLField(max_length=100)
-                custom_field = ModelField(model_field=<tests.test_model_serializer.CustomField: custom_field>)
-                file_path_field = FilePathField(path=%r)
+        expected = dedent(r"""
```

Comment:
usage of regex instead of old string format seems interesting to me. can you share more insight about it? pros and cons regarding this PR?

Reply:
The need arises from the integer validator generated from  Django 5 integer fields. That validator breaks the assertion against the repr. In other tests, there is a conditional on Django version, an "if" to add the extra string to match. I find regex to be more flexible against changes that are not the object of the test - i.e. the correct functionality asserted from the repr in which the integer validator does not play a role. This would also prevent cluttering if, in future versions of Django, similar changes arrive (preventing the discussion for each version).

The cons I find would be the escaping cluttering in the strings because otherwise, it is pretty straight with the current testing suite ("the dedent of strings for testing the repr" way) and not harder to read.

Reply:
I'm reading this as "I like things complicated". 🙃